src/quo/components/buttons/button/style.cljs

```diff
@@ -55,61 +56,64 @@
-                                  32 5
-                                  24 0
-                                  9))
-          :overflow           :hidden
-          :background-color   background-color
-          :border-radius      (if border-radius
-                                border-radius
-                                (case size
-                                  56 12
-                                  40 12
-                                  32 10
-                                  24 8
-                                  12))
-          :border-color       border-color
-          :border-width       (when border-color 1)}
-         (when icon-only?
-           {:width size})
-         (when border-color
-           {:border-color border-color
-            :border-width 1})
-         (when disabled?
-           {:opacity 0.3})))
+  (let [disabled-and-rgb-hex? (and disabled? (colors/rgb-hex? background-color))]
+    (merge {:height             size
+            :align-items        :center
+            :justify-content    :center
+            :flex-direction     (if icon-top :column :row)
+            :padding-horizontal (when-not (or icon-only? icon-left icon-right)
+                                  (case size
+                                    56 (if border-color 10 11)
+                                    40 16
+                                    32 12
+                                    24 7
+                                    16))
+            :padding-left       (when-not (or icon-only? icon-left)
+                                  (case size
+                                    56 nil
+                                    40 16
+                                    32 12
+                                    24 8
+                                    16))
+            :padding-right      (when-not (or icon-only? icon-right)
+                                  (case size
+                                    56 nil
+                                    40 16
+                                    32 12
+                                    24 8
+                                    16))
+            :padding-top        (when-not (or icon-only? icon-left icon-right)
+                                  (case size
+                                    56 0
+                                    40 (if border-color 8 9)
+                                    32 (if border-color 4 5)
+                                    24 0
+                                    (if border-color 8 9)))
+            :padding-bottom     (when-not (or icon-only? icon-left icon-right)
+                                  (case size
+                                    56 0
+                                    40 9
+                                    32 5
+                                    24 0
+                                    9))
+            :overflow           :hidden
+            :background-color   (if disabled-and-rgb-hex?
```

Comment:
why do we need to check if it is `hex` as well? - can't we just check if it is disabled and app the opacity of 0.3 if so?

Reply:
```
:background-color (if disabled? (colors/alpha background-color 0.3)  background-color)
```

Reply:
Thank you so much Jamie, I totally forgot the we had an alpha function in the colors ns :)
Removing the conversion I created :)

Reply:
The check was to make sure the hex color is only of 7 digits (Hex color without opacity value.)